### PR TITLE
POS Blockhash Compatibility

### PR DESCRIPTION
**Pull request by beether**
In PoW, miners cannot easily craft a desired `blockhash` without incurring an opportunity cost for missing the block (they will likely only get an uncle reward, and not the full block reward). This behavior should arguably be maintained in PoS so that a source of randomness is available, albeit with certain limitations.

### CODE REVIEW

**Comment by MicahZoltu** [comment]
While I agree with the sentiment, it is not immediately obvious to me that this is possible.  I think in order for this to be merged as as a draft EIP it needs to propose at actually solution to the problem of how to generate `entropyhash`.

You may want to start a discussion over at ethresear.ch to see if others can help you figure out how to make it random.

**Comment by beether** [comment]
@MicahZoltu Do you happen to know by what mechanism PoS decides which validator wins the block, or where I could read about it?

**Comment by MicahZoltu** [comment]
I don't personally know how that process works in enough detail to explain it or speak intelligently about it.  ethresear.ch is where most Casper conversation is happening these days though, so I recommend reading there.  I believe there is an entire section just for PoS.

**Comment by beether** [comment]
@MicahZoltu Thanks. I've created a topic. We'll see how it goes.

File: EIPS/eip-pos-blockhash.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+## Preamble
```
Comment (Arachnid) [inline EIPS/eip-pos-blockhash.md]: Please update this to the current header format described in EIP-X and EIP-1.

File: EIPS/eip-pos-blockhash.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+---
+eip: <to be assigned>
```
Comment (Arachnid) [inline EIPS/eip-pos-blockhash.md]: Please number this 1023 and rename to eip-1023.md

File: EIPS/eip-pos-blockhash.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+title: PoS blockhash backward compatibility
+author: Beether
+type: Standard Track
```
Comment (Arachnid) [inline EIPS/eip-pos-blockhash.md]: This should be 'Standards Track'

**Comment by jamesray1** [comment]
FYI: https://ethresear.ch/t/rng-exploitability-analysis-assuming-pure-randao-based-main-chain/1825/5?u=jamesray1/

File: EIPS/eip-1023.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+---
+eip: 1023
+title: PoS blockhash backward compatibility
+author: Beether
```
Comment (Arachnid) [inline EIPS/eip-1023.md]: Please provide a github username or email address.

File: EIPS/eip-1023.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+## Specification
+Let `entropyhash` for a particuar `blockheight` be a value unknown to any validator before their block and vote are cast. This ensures a validator has to commit to their _possibly-malicious-transactions_ before knowing what the results will be.
+
+I'm not a PoS expert, so I don't know how exactly to generate `entropyhash`, but it seems that hashing `data-used-to-pick-winning-validator` is what we're after. No validator shoud know this ahead of time, and the result of this value should have an economic stake behind it.
```
Comment (Arachnid) [inline EIPS/eip-1023.md]: This is much too vague to make an actual standard. If you're looking for ideas, I'd suggest posting your question to the ethereum-magicians forum instead.

**Comment by axic** [comment]
@beether are you still pursuing this?

**Comment by axic** [comment]
If I understand this proposal correctly:
- it reasons that blockhash is a good source of randomness
- argues that in PoS it may not be a good source
- proposes to introduce some kind of randomness feature

I have to agree with @Arachnid that this likely is a discussion relevant to https://ethresear.ch or https://ethereum-magicians.org and suggest to close this PR.

**Pull request closed without merging**